src/test/java/com/hyperlocal/server/ShopControllerTest.java

```diff
@@ -43,6 +56,150 @@
   @Mock
   ResultSet resultSet;
 
+  @Test
+  public void shouldGetShopsByMerchantID() throws Exception {
+    assertThat(controller).isNotNull();
+    
+    // Param
+    Long merchantID = 1000000000000L;
+
+    // Def: Mocked database output
+    RowData shopRecord = new Record(1L, 200L, "Arvind Shop", new BigDecimal(23.33), new BigDecimal(23.33), "Mumbai", "Groceries");
+    ResultSet shopRecords = new Records(shopRecord);
+    QueryResult shopsQueryResult = new QueryResult(0L, "Success", shopRecords);
+
+    CompletableFuture<QueryResult> queryResultPromise = CompletableFuture.completedFuture(shopsQueryResult);
+    when(connection.sendPreparedStatement(SELECT_SHOPS_BY_MERCHANT_STATEMENT, Arrays.asList(merchantID)))
+    .thenReturn(queryResultPromise);
+
+    // Def: Expected output:
+    HashMap<String, Object> expectedMap = new HashMap<String, Object>();
+    expectedMap.put("count", 1);
+    expectedMap.put("shops", new ArrayList<Shop>(Arrays.asList(new Shop(shopRecord))));
+
+    
+    // Verify: Expected vs actual output
+    CompletableFuture<HashMap<String, Object>> actualMapPromise = controller.getShopsByMerchantID(merchantID);
+    assertEquals(expectedMap, actualMapPromise.get());
+    verify(connection).sendPreparedStatement(SELECT_SHOPS_BY_MERCHANT_STATEMENT, Arrays.asList(merchantID));
+  }
```

Comment:
It is generally better to arrange your test cases in AAA form

// Arrange
All your mocks, initializations etc


// Act
Call the method that you want to test


// Assert
check for correctness

Reply:
https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80

Reply:
On it. thanks!